models/DataObject/ClassDefinition/Data.php

```diff
@@ -1208,6 +1220,21 @@ public function getUnique()
      * @throws \Exception
      */
     protected function getDataFromObjectParam($object, $params = [])
+    {
+        return $this->getSetData($object, $params);
+    }
+
+    /**
+     * @param DataObject\Concrete|DataObject\Localizedfield|DataObject\Objectbrick\Data\AbstractData|DataObject\Fieldcollection\Data\AbstractData $object
+     * @param array $params
+     * @param mixed|null $setData
+     * @param string|null method
+     *
+     * @return mixed
+     *
+     * @throws \Exception
+     */
+    protected function getSetData($object,  $params = [], $setData = null, $method = 'get')
```

Comment:
Should we maybe split this into 2 dedicated methods for better readability? 
For me it's seems to be quite hard to read the code.

Reply:
I second that, that is super hard to read and also to understand....

Reply:
I was just trying to save some duplication but yes, it leads to poor readability... so changed it :)